android/greengrass-nucleus-android/src/main/java/com/aws/greengrass/android/managers/ServicesConfigurationProvider.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.android.managers;
+
+import com.aws.greengrass.android.provision.WorkspaceManager;
+import lombok.NonNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
+public class ServicesConfigurationProvider {
+    private static ServicesConfigurationProvider instance = null;
+    WorkspaceManager workspace = null;
+
+    private ServicesConfigurationProvider(@NonNull File filesDir) {
+        workspace = WorkspaceManager.getInstance(filesDir);
+    }
+
+    /**
+     * Gets instance of ServicesConfigurationProvider.
+     * @param filesDir path to application's files/ directory
+     * @return instance of ServicesConfigurationProvider
+     */
+    public static synchronized ServicesConfigurationProvider getInstance(@NonNull File filesDir) {
+        if (instance == null) {
+            instance = new ServicesConfigurationProvider(filesDir);
+        }
+        return  instance;
+    }
+
+    /**
+     * Provides external services config to Nucleus.
+     *
+     * @param is external config input stream
+     */
+    public void setExternalConfig(InputStream is) {
+        File destinationFile = Paths.get(workspace.getConfigPath().toString(), "config.yaml").toFile();
+        try {
+            if (!destinationFile.exists()) {
+                destinationFile.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)